### PR TITLE
154 e2e replace cypress with playwright

### LINKED ISSUE #154: E2E, replace cypress with Playwright
https://nx.dev/nx-api/playwright

### PR DESCRIPTION
**Pull request by AlexandreG-tech**
## PR Type:
Refactoring

___
## PR Description:
This PR mainly focuses on two aspects:
- Replacing the end-to-end testing framework Cypress with Playwright.
- Implementing currency conversion for Stripe payments.

Other changes include:
- Updating the file upload library from 'upload-js-full' to '@bytescale/sdk'.
- Refactoring the error handling for failed NFT claims.
- Updating the JWT generation method.
- Minor changes in several files for code cleanup and optimization.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

- `libs/payment/admin/src/lib/payment-admin.ts`: Implemented currency conversion for Stripe payments using the 'CurrencyCache' and 'toUserCurrency' utilities. The currency conversion is applied when creating the line items for the Stripe checkout session.
- `libs/payment/admin/src/lib/payment-admin.spec.ts`: Updated the tests to reflect the changes in the 'Payment' class, especially the currency conversion and error handling for failed NFT claims.
- `libs/features/back-office/events/src/lib/EventCards.tsx`: Replaced the 'react-uploader' with '@bytescale/upload-widget-react' for file uploads. Also, removed the 'uploader' prop from the 'EventCard' component.
- `libs/payment/webhooks/src/lib/payment-webhooks.spec.ts`: Updated the tests to reflect the changes in the 'stripeCheckoutStatus' function, especially the error handling for failed NFT claims.
- `libs/file-upload/admin/src/lib/index.ts`: Updated the file upload library from 'upload-js-full' to '@bytescale/sdk'. Also, replaced the 'FileCopyStatus' enum with 'FileCopyStatusEnum'.
- `libs/test-utils/gql/src/index.ts`: Updated the JWT generation method to use 'jsonwebtoken' instead of 'jwt'. Also, updated the 'applicantId' for the 'alpha_user' account.
- `libs/features/pass-api/src/lib/revealPass.spec.ts`: Updated the tests to reflect the changes in the 'revealPass' function, especially the error handling for failed file copies.
- `libs/file-upload/admin/src/lib/index.spec.ts`: Updated the tests to reflect the changes in the 'FileWrapper' and 'UploadWrapper' classes, especially the initialization of the 'UploadManager' and 'FileApi'.
- `libs/features/pass-api/src/lib/transferPassQrCode.spec.ts`: Updated the tests to reflect the changes in the 'transferPassQrCode' function, especially the error handling for failed file copies.
</details>

### CODE REVIEW

File: apps/web/playwright/utils/loadUser.ts
Comment (sebpalluel) [inline apps/web/playwright/utils/loadUser.ts:1]: Reprendre les données ('id', 'address', 'email') des accounts existants de `@test-utils/gql`:
```
export const accounts = {
  google_user: {
    id: 'ac542c34-1907-451c-94be-5df69a959080',
    address: '0x1bBEdB07706728A19c9dB82d3c420670D8040592',
    email: 'googl_user@gmail.com',
  } satisfies AppUser,
  alpha_user: {
    id: '679f92d6-a01e-4ab7-93f8-10840d22b0a5',
    address: '0xB98bD7C7f656290071E52D1aA617D9cB4467Fd6D',
    email: 'alpha_user@test.io',
    kyc: {
      applicantId: '64d5f7a58f241166d756ba33',
      reviewStatus: KycStatus_Enum.Completed,
      levelName: KycLevelName_Enum.BasicKycLevel,
    },
  } satisfies AppUser,
  beta_user: {
    id: '76189546-6368-4325-8aad-220e03837b7e',
    address: '0x1B8bD7C7f656290071E52D1aA617D9cB4469BB9F',
  } satisfies AppUser,
};
```

File: libs/next/auth/src/lib/authProvider.tsx [hunk 1]
```diff
@@ -38,12 +40,39 @@ interface AuthContextValue {
 
 const AuthContext = createContext<AuthContextValue | undefined>(undefined);
 
+declare global {
```
Comment (sebpalluel) [inline libs/next/auth/src/lib/authProvider.tsx]: Doit être define ici `libs/next/types/src/lib/window.ts`
Puis ajouter `"types": ["@next/types"]` dans `libs/next/auth/tsconfig.json`

éventuellement ajouter ceci si erreur de build:
```
// eslint-disable-next-line import/no-unresolved
import '@next/types';
```

File: libs/next/i18n/src/messages/fr.json [hunk 2]
```diff
@@ -103,6 +103,7 @@
       "continue-shopping-button": "Aller à la page d'accueil"
     },
     "Cancelled": {
+      "title": "Achat annulé",
       "description": "Votre achat a été annulé avec succès. Nous sommes désolés de vous voir partir. Si vous avez des questions ou des préoccupations, n'hésitez pas à nous contacter. N'hésitez pas à continuer vos achats si vous changez d'avis.",
```
Comment (sebpalluel) [inline libs/next/i18n/src/messages/fr.json:107]: title est en double

File: libs/payment/admin/src/lib/payment-admin.ts [hunk 3]
```diff
@@ -219,12 +226,34 @@ export class Payment {
...
       }
+      let currencyStripe: string;
+      let unitAmount: number;
+
+      if (currency === order.eventPassPricing.priceCurrency) {
+        currencyStripe = currency.toLowerCase();
+        unitAmount = order.eventPassPricing.priceAmount;
+      } else {
+        currencyStripe = currency.toLowerCase();
+        unitAmount = Math.floor(
```
Comment (sebpalluel) [inline libs/payment/admin/src/lib/payment-admin.ts]: Test ! et extraction en methode avec nom approprié dans `@next/currency-common`

File: libs/payment/admin/src/lib/payment-admin.ts [hunk 4]
```diff
@@ -257,12 +286,13 @@ export class Payment {
...
       );
     }
+    const lineItems = await Promise.all(lineItemsPromises);
 
     const session = await this.stripe.checkout.sessions.create({
       line_items: lineItems,
       client_reference_id: user.id,
       customer: stripeCustomer.id,
-      currency,
+      currency: currency.toLowerCase(),
```
Comment (sebpalluel) [inline libs/payment/admin/src/lib/payment-admin.ts:296]: pq pas currencyStripe ?

File: libs/test-utils/gql/src/index.ts [hunk 5]
```diff
@@ -76,8 +77,13 @@ const H256SecretBackOffice = 'HGTRbJ6IaEoByH8KhA+BKV0Bgug+R7RSydnMbex2cZg=';
 // generate a JWT that includes roles, userId
 // secret: private key provided on docker-compose for test, either for marketplace or back-office
 const generateJwt = (user: AppUser, secret: string, role?: string): string =>
-  jwt.sign(JSON.stringify({ user, role }), secret);
-
+  jsonwebtoken.sign(
+    JSON.stringify({ user, role }),
+    process.env.NEXTAUTH_SECRET || 'default_secret',
```
Comment (sebpalluel) [inline libs/test-utils/gql/src/index.ts]: Ou c'est utilisé  `default_secret` ? J'imagine que c'est pour bypass le process.env.NEXTAUTH_SECRET non provide en e2e ?

File: libs/ui/components/src/lib/dropdown-menu/DropdownMenu.tsx [hunk 6]
```diff
@@ -83,10 +83,9 @@ const DropdownMenuItem = React.forwardRef<
   React.ElementRef<typeof DropdownMenuPrimitive.Item>,
```
Comment (sebpalluel) [inline libs/ui/components/src/lib/dropdown-menu/DropdownMenu.tsx]: Je préfère qu'on garde cette version plus DRY

File: libs/ui/components/src/lib/dropdown-menu/DropdownMenuItemClient.tsx [hunk 7]
```diff
@@ -39,7 +39,7 @@ export interface DropdownMenuItemClientProps {
 }
 export const DropdownMenuItemClient: React.FC<DropdownMenuItemClientProps> = ({
   icon,
-  item,
+  item: { wrapper, ...item },
   iconClasses,
   setLoading,
```
Comment (sebpalluel) [inline libs/ui/components/src/lib/dropdown-menu/DropdownMenuItemClient.tsx]: Du coup rollback là dessus et laisser DropdownMenuItem gérer le wrapper

File: package.json [hunk 8]
```diff
@@ -131,7 +137,6 @@
     "tailwindcss-animate": "^1.0.7",
     "tailwindcss-gradients": "^3.0.0",
     "tslib": "^2.6.2",
```
Comment (sebpalluel) [inline package.json:139]: Supprimer tous les packages cypress

**Review by sebpalluel — COMMENTED** [review]
Où est la suppression de tout le code concernant cypress maintenant deprecated ?

File: libs/payment/admin/src/lib/payment-admin.ts (hunk 4 above)
Comment (AlexandreG-tech) [reply libs/payment/admin/src/lib/payment-admin.ts:296]: Question de scope comme currencyStripe est crée dans le mapping d'une ordre

**Review by sebpalluel — APPROVED** [review]

**Review by sebpalluel — APPROVED** [review]

**Review by sebpalluel — APPROVED** [review]

**Review by sebpalluel — APPROVED** [review]

**Review by sebpalluel — APPROVED** [review]

**Review by sebpalluel — APPROVED** [review]

**Review by sebpalluel — APPROVED** [review]

**Pull request merged**